cmd/opampsupervisor/e2e_test.go

```diff
@@ -2337,8 +2337,12 @@ func TestSupervisorHealthCheckServerBackendConnError(t *testing.T) {
 	healthcheckPort, err := findRandomPort()
 	require.NoError(t, err)
 
+	// Find an open port on the host that has no server listening on it.
+	badOpAMPServerPort, err := findRandomPort()
+	require.NoError(t, err)
+
 	cfgFile := getSupervisorConfig(t, "healthcheck", map[string]string{
-		"url":      "badserver:8080",
+		"url":      fmt.Sprintf("localhost:%d", badOpAMPServerPort),
```

Comment:
This is one key fix: the DNS lookup for an unknown host name apparently cannot be cancelled, and therefore was producing the leaking goroutine. I couldn't figure out how to fix that issue (and it likely shouldn't be a problem when running the Supervisor as a process), so instead I opted to just avoid it.